signcrypt_seal.go

```diff
@@ -250,6 +240,24 @@ func (sss *signcryptSealStream) init(receiverBoxKeys []BoxPublicKey, receiverSym
 	return nil
 }
 
+func sealEncryptionKeyForReceivers(receiverBoxKeys []BoxPublicKey, receiverSymmetricKeys []ReceiverSymmetricKey, ephemeralKey BoxSecretKey, encryptionKey SymmetricKey) []receiverKeys {
+	order := randomPerm(len(receiverBoxKeys) + len(receiverSymmetricKeys))
+
+	receiverKeysArray := make([]receiverKeys, len(receiverBoxKeys)+len(receiverSymmetricKeys))
+	// Collect all the recipient identifiers, and encrypt the payload key for
+	// all of them.
+	for i, receiverBoxKey := range receiverBoxKeys {
+		index := order[i]
+		receiverKeysArray[index] = receiverEntryForBoxKey(receiverBoxKey, ephemeralKey, encryptionKey, uint64(index))
+	}
+	for i, receiverSymmetricKey := range receiverSymmetricKeys {
+		index := order[len(receiverBoxKeys)+i]
+		receiverKeysArray[index] = receiverEntryForSymmetricKey(receiverSymmetricKey, ephemeralKey.GetPublicKey(), encryptionKey, uint64(index))
+	}
```

Comment:
Would it be easier to try to define a function that permuted an array in place? Even if we had to copy it for different data types?